landing-pages: Add section on chat.zulip.org custom linkifiers.

laurynmm opened the pull request:
Adds a sub-section in the Zulip [development community](https://zulip.com/development-community/) landing page about using the custom linkifiers in chat.zulip.org for the different Zulip project repositories on GitHub.

See [relevant conversation in CZO](https://chat.zulip.org/#narrow/stream/137-feedback/topic/linkifier.20visibility/near/1513440).

---

**Screenshots and screen captures:**

<details>
<summary>New linkifiers section</summary>

![Screenshot from 2023-03-07 13-36-52](https://user-images.githubusercontent.com/63245456/223427223-69c70ab5-3d28-496c-8dbd-3edb9554916f.png)
</details>
<details>
<summary>Updated "Anything else to keep in mind?" section</summary>

[Current page](https://zulip.com/development-community/#anything-else-to-keep-in-mind)
![Screenshot from 2023-03-07 13-46-51](https://user-images.githubusercontent.com/63245456/223427208-5ac1c7a9-9472-41e6-bca5-739a21abc9fd.png)
</details>

---

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented on templates/corporate/development-community.md:
```diff
@@ -226,12 +226,30 @@ This will search the full history of all public streams for `<your
+
+We've set up [custom linkifiers](/help/add-a-custom-linkifier) so that
+it's easy to link to issues and pull requests for the various Zulip
+project repositories on GitHub when composing a message or [creating a
```
How about:

> ... it's easy to link to issues and pull requests in Zulip
project repositories on GitHub when composing a message or [starting a new topic](/help/starting-a-new-topic).

alya commented on templates/corporate/development-community.md:
```diff
@@ -226,12 +226,30 @@ This will search the full history of all public streams for `<your
+- the [Zulip mobile client][mobile-repository] repository, use `#M1234`.
+- the [Zulip terminal client][terminal-repository] repository, use `#T1234`.
+- the [Zulip desktop client][desktop-repository] repository, use `#D1234`.
+
```
I wonder if a table would work better here?

laurynmm replied:
Maybe? But we'd need to figure out some CSS styling. I think the only other table in the landing pages is on the plans page where we compare Zulip to other applications, and that formatting wouldn't work. We definitely don't have any tables in the markdown files for the landing pages.

<details>
<summary>Rough draft of reformatting markdown as a table</summary>

![Screenshot from 2023-03-07 20-12-08](https://user-images.githubusercontent.com/63245456/223528168-8a0e63de-db7f-434b-bb18-8adcd4f1a389.png)
</details>

laurynmm replied:
Maybe we could format the lead-in phrase to the bullet points better so the points could just be:
- Repository name: `linkifier text`

timabbott replied:
Looks nice to me; I don't think we need a Markdown table.

laurynmm commented:
**Updates**:
- Changed the 'it's easy ...' text for updated text above.
- Updated bullet points to be simpler and have less repetitive text

**Updated screenshots**:

<details>
<summary>Linking to Github ... section</summary>

![Screenshot from 2023-03-08 13-00-07](https://user-images.githubusercontent.com/63245456/223709253-4b66d61f-403c-4058-aa1d-d2947927974b.png)
</details>
<details>
<summary>Anything else to keep in mind? section</summary>

![Screenshot from 2023-03-08 13-00-19](https://user-images.githubusercontent.com/63245456/223709241-1862fef7-8ef4-46c2-8f74-f6cb83a0dddf.png)
</details>

alya commented:
Works for me!

timabbott commented:
Looks great, merged, thanks @laurynmm!
